kore/src/Kore/Step/Axiom/EvaluationStrategy.hs

```diff
@@ -336,3 +351,50 @@ evaluateWithDefinitionAxioms
         $ Step.applyRulesSequence unificationProcedure initial rules
 
     unificationProcedure = UnificationProcedure unificationWithAppMatchOnTop
+
+introduceDefinedness
+    :: forall variable
+    .  ( SortedVariable variable
+       , FreshVariable variable
+       , Unparse variable
+       , Show variable
+       )
+    => OrPredicate variable
+    -> Pattern variable
+    -> Pattern variable
+introduceDefinedness cond result =
+    foldl andCondition result cond
+
+ceilChildOfApplication
+    :: forall variable simplifier
+    .  ( FreshVariable variable
+       , SortedVariable variable
+       , Show variable
+       , Unparse variable
+       , MonadSimplify simplifier
+       )
+    => TermLike variable
+    -> simplifier (OrPredicate variable)
+ceilChildOfApplication =
+    withCeilChildOfApplication
+        Ceil.makeEvaluateTerm
+        AndPredicates.simplifyEvaluatedMultiPredicate
+
+withCeilChildOfApplication
```

Comment:
Let's move the three new functions in this module into `Kore.Step.Remainder` because they are all related to building the remainder.